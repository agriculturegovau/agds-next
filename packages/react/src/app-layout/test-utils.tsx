import { Fragment } from 'react';
import { Text } from '../text';
import { NotificationBadge } from '../notification-badge';
import { VisuallyHidden } from '../a11y';
import {
	ChartLineIcon,
	EmailIcon,
	ExitIcon,
	HelpIcon,
	HomeIcon,
	SuccessIcon,
	SettingsIcon,
	FactoryIcon,
	AvatarIcon,
	ChevronsLeftIcon,
	MinusIcon,
} from '../icon';
import {
	DropdownMenuDivider,
	DropdownMenuItem,
	DropdownMenuItemLink,
	DropdownMenuPanel,
	DropdownMenuGroup,
	DropdownMenuGroupLink,
	DropdownMenuItemRadio,
} from '../dropdown-menu';

export const navigationItems = (businessName: string) => [
	{
		options: { disableGroupPadding: true },
		items: [
			{
				label: 'Back to my account',
				icon: ChevronsLeftIcon,
				href: '/account',
			},
		],
	},
	{
		options: { disableGroupPadding: true },
		items: [
			{
				label: (
					<Fragment>
						<Text fontWeight="bold" fontSize="xs">
							{businessName}
						</Text>
						<Text color="muted" fontSize="xs">
							ABN: 00 000 000 000
						</Text>
					</Fragment>
				),
			},
		],
	},
	[
		{
			label: 'Dashboard',
			icon: HomeIcon,
			href: '/',
		},
		{
			label: 'Establishments',
			icon: FactoryIcon,
			href: '/establishments',
			items: [
				{
					label: 'Canberra',
					icon: MinusIcon,
					href: '/establishments/canberra',
				},
				{
					label: 'Sydney',
					icon: MinusIcon,
					href: '/establishments/sydney',
				},
			],
		},
		{
			label: 'Data and Insights',
			icon: ChartLineIcon,
			href: '/intelligence',
		},
		{
			label: 'Compliance',
			icon: SuccessIcon,
			href: '/compliance',
			items: [
				{
					label: 'Audit',
					icon: MinusIcon,
					href: '/compliance/audit',
				},
				{
					label: 'Certificates',
					icon: MinusIcon,
					href: '/compliance/certificates',
				},
			],
		},
	],

	[
		{
			label: 'Help',
			icon: HelpIcon,
			href: '/help',
		},
	],
	[
		{
			label: 'Sign out',
			onClick: console.log,
			icon: ExitIcon,
		},
	],
];

export const exampleData = {
	userNames: {
		short: 'Ab Cd',
		regular: 'Toto Wolff',
		medium: 'Benjamin Harrington',
		long: 'Benjamin Alexander Harrington-Smythe',
	},
	businessNames: {
		short: ['Lorem', 'Ipsum', 'Company'],
		regular: ['Antfix', 'Produce Fresh', 'Organic Co'],
		medium: [
			'Malesuada lacus a tortor blandit fermentum phasellus',
			'Iaculis tortor duis ante nec risus elementum id ui',
			'Purus tortor lacus malesuada phasellus ipsum ex duis libero ante id',
		],
		long: [
			'Duis quis iaculis libero nec consectetur luctus ex in malesuada lacus a tortor blandit fermentum phasellus iaculis ipsum nec purus aliquet id elementum risus duis ante ui',
			'Duis ut a libero duis nisl ex lacus in luctus viverra iaculis iaculis tortor duis malesuada blandit purus diam ipsum ante nec risus elementum id ui nec quis aliquet phasellus fermentum',
			'Blandit iaculis iaculis quis ante diam viverra elementum ui risus nec luctus purus tortor lacus malesuada phasellus ipsum ex duis libero ante id Duis fermentum nisl aliquet duis in a nec ut consectetur',
		],
	},
};

export function ExampleAccountDropdown({
	businesses = exampleData.businessNames.regular,
	onBusinessChange,
	selectedBusinessName,
}: {
	businesses?: string[];
	onBusinessChange: (value: string) => void;
	selectedBusinessName: string;
}) {
	return (
		<DropdownMenuPanel palette="light">
			<DropdownMenuGroup label="Businesses">
				{businesses.map((businessName) => (
					<DropdownMenuItemRadio
						key={businessName}
						checked={businessName === selectedBusinessName}
						secondaryText="ABN 00 000 000 000"
						onClick={() => onBusinessChange?.(businessName)}
					>
						{businessName}
					</DropdownMenuItemRadio>
				))}
				<DropdownMenuGroupLink href="#">View all</DropdownMenuGroupLink>
			</DropdownMenuGroup>
			<DropdownMenuDivider />
			<DropdownMenuGroup label="My account">
				<DropdownMenuItemLink href="/profile" icon={AvatarIcon}>
					Profile
				</DropdownMenuItemLink>
				<DropdownMenuItemLink
					href="/messages"
					icon={EmailIcon}
					endElement={
						<span>
							<NotificationBadge tone="action" value={6} max={99} aria-hidden />
							<VisuallyHidden>, 6 unread</VisuallyHidden>
						</span>
					}
				>
					Messages
				</DropdownMenuItemLink>
				<DropdownMenuItemLink href="/account-settings" icon={SettingsIcon}>
					Account settings
				</DropdownMenuItemLink>
			</DropdownMenuGroup>
			<DropdownMenuDivider />
			<DropdownMenuItem onClick={() => console.log('sign out')} icon={ExitIcon}>
				Sign out
			</DropdownMenuItem>
		</DropdownMenuPanel>
	);
}
