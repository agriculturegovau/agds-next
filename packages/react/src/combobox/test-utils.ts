export type Option = { label: string; value: string };

export const COUNTRY_OPTIONS = [
	'Afghanistan is a really long country that spans multiple lines cmon this can be way longer than that Afghanistan is a really long country that spans multiple lines cmon this can be way longer than that',
	'Albania',
	'Algeria',
	'Andorra',
	'Angola',
	'Antigua and Barbuda',
	'Argentina',
	'Armenia',
	'Australia',
	'Austria',
	'Azerbaijan',
	'Bahamas',
	'Bahrain',
	'Bangladesh',
	'Barbados',
	'Belarus',
	'Belgium',
	'Belize',
	'Benin',
	'Bhutan',
	'Bolivia',
	'Bosnia and Herzegovina',
	'Botswana',
	'Brazil',
	'Brunei',
	'Bulgaria',
	'Burkina Faso',
	'Burundi',
	"CÃ´te d'Ivoire",
	'Cabo Verde',
	'Cambodia',
	'Cameroon',
	'Canada',
	'Central African Republic',
	'Chad',
	'Chile',
	'China',
	'Colombia',
	'Comoros',
	'Congo (Congo-Brazzaville)',
	'Costa Rica',
	'Croatia',
	'Cuba',
	'Cyprus',
	'Czechia (Czech Republic)',
	'Democratic Republic of the Congo',
	'Denmark',
	'Djibouti',
	'Dominica',
	'Dominican Republic',
	'Ecuador',
	'Egypt',
	'El Salvador',
	'Equatorial Guinea',
	'Eritrea',
	'Estonia',
	'Eswatini (fmr. "Swaziland")',
	'Ethiopia',
	'Fiji',
	'Finland',
	'France',
	'Gabon',
	'Gambia',
	'Georgia',
	'Germany',
	'Ghana',
	'Greece',
	'Grenada',
	'Guatemala',
	'Guinea',
	'Guinea-Bissau',
	'Guyana',
	'Haiti',
	'Holy See',
	'Honduras',
	'Hungary',
	'Iceland',
	'India',
	'Indonesia',
	'Iran',
	'Iraq',
	'Ireland',
	'Israel',
	'Italy',
	'Jamaica',
	'Japan',
	'Jordan',
	'Kazakhstan',
	'Kenya',
	'Kiribati',
	'Kuwait',
	'Kyrgyzstan',
	'Laos',
	'Latvia',
	'Lebanon',
	'Lesotho',
	'Liberia',
	'Libya',
	'Liechtenstein',
	'Lithuania',
	'Luxembourg',
	'Madagascar',
	'Malawi',
	'Malaysia',
	'Maldives',
	'Mali',
	'Malta',
	'Marshall Islands',
	'Mauritania',
	'Mauritius',
	'Mexico',
	'Micronesia',
	'Moldova',
	'Monaco',
	'Mongolia',
	'Montenegro',
	'Morocco',
	'Mozambique',
	'Myanmar (formerly Burma)',
	'Namibia',
	'Nauru',
	'Nepal',
	'Netherlands',
	'New Zealand',
	'Nicaragua',
	'Niger',
	'Nigeria',
	'North Korea',
	'North Macedonia',
	'Norway',
	'Oman',
	'Pakistan',
	'Palau',
	'Palestine State',
	'Panama',
	'Papua New Guinea',
	'Paraguay',
	'Peru',
	'Philippines',
	'Poland',
	'Portugal',
	'Qatar',
	'Romania',
	'Russia',
	'Rwanda',
	'Saint Kitts and Nevis',
	'Saint Lucia',
	'Saint Vincent and the Grenadines',
	'Samoa',
	'San Marino',
	'Sao Tome and Principe',
	'Saudi Arabia',
	'Senegal',
	'Serbia',
	'Seychelles',
	'Sierra Leone',
	'Singapore',
	'Slovakia',
	'Slovenia',
	'Solomon Islands',
	'Somalia',
	'South Africa',
	'South Korea',
	'South Sudan',
	'Spain',
	'Sri Lanka',
	'Sudan',
	'Suriname',
	'Sweden',
	'Switzerland',
	'Syria',
	'Tajikistan',
	'Tanzania',
	'Thailand',
	'Timor-Leste',
	'Togo',
	'Tonga',
	'Trinidad and Tobago',
	'Tunisia',
	'Turkey',
	'Turkmenistan',
	'Tuvalu',
	'Uganda',
	'Ukraine',
	'United Arab Emirates',
	'United Kingdom',
	'United States of America',
	'Uruguay',
	'Uzbekistan',
	'Vanuatu',
	'Venezuela',
	'Vietnam',
	'Yemen',
	'Zambia',
	'Zimbabwe',
].map((country) => ({ label: country, value: country }));

export const STATE_OPTIONS = [
	{ label: 'Australian Capital Territory', value: 'act' },
	{ label: 'New South Wales', value: 'nsw' },
	{ label: 'Northern Territory', value: 'nt' },
	{ label: 'Queensland', value: 'qld' },
	{ label: 'South Australia', value: 'sa' },
	{ label: 'Tasmania', value: 'tas' },
	{ label: 'Victoria', value: 'vic' },
	{ label: 'Western Australia', value: 'wa' },
];

export const NAME_OPTIONS = [
	{
		fullName: 'John Doe',
		jobTitle: 'Product Manager',
		status: 'Draft',
		unreadMessageCount: 0,
	},
	{
		fullName: 'Jane Smith',
		jobTitle: 'Software Engineer',
		status: 'Approved',
		unreadMessageCount: 4,
	},
	{
		fullName: 'Robert Johnson',
		jobTitle: 'UX Designer',
		status: 'Deactivated',
		unreadMessageCount: 5,
	},
	{
		fullName: 'Emily Davis',
		jobTitle: 'Product Manager',
		status: 'Approved',
		unreadMessageCount: 6,
	},
	{
		fullName: 'Michael Williams',
		jobTitle: 'Software Engineer',
		status: 'Approved',
		unreadMessageCount: 0,
	},
	{
		fullName: 'Sarah Wilson',
		jobTitle: 'UX Designer',
		status: 'Approved',
		unreadMessageCount: 0,
	},
	{
		fullName: 'David Brown',
		jobTitle: 'Product Manager',
		status: 'Deactivated',
		unreadMessageCount: 0,
	},
	{
		fullName: 'Olivia Taylor',
		jobTitle: 'Software Engineer',
		status: 'Approved',
		unreadMessageCount: 6,
	},
	{
		fullName: 'James Moore',
		jobTitle: 'UX Designer',
		status: 'Approved',
		unreadMessageCount: 0,
	},
	{
		fullName: 'Sophia Clark',
		jobTitle: 'Product Manager',
		status: 'Approved',
		unreadMessageCount: 0,
	},
].map((option) => ({
	...option,
	label: option.fullName,
	value: option.fullName,
}));
