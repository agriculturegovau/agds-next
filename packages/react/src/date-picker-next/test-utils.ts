import * as yup from 'yup';

export const errorMessage = 'Enter a valid date';

// `yup.date()` can sometimes give false positives with certain string values
// Fixes https://github.com/jquense/yup/issues/764
// Tests for this can be viewed in the tests for `DatePicker` and `DateRangePicker`
// IMPORTANT: This is for example purposes only. It should NOT be exported from the design system
export const yupDateField = yup
	.date()
	.transform((current, orig) => (typeof orig === 'string' ? orig : current))
	.typeError(errorMessage);
