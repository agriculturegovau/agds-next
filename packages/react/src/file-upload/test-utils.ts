import { FileStatus, FileWithStatus } from './utils';

export function createExampleFileWithStatus(status?: FileStatus) {
	const bits = ['example'];
	const file: FileWithStatus = new File(bits, 'example.txt', {
		type: 'text/plain',
	});
	file.status = status;
	return file;
}

export function createExampleImageFileWithStatus(status?: FileStatus) {
	const file: FileWithStatus = new File(['example'], 'example.jpg', {
		type: 'image/jpg',
	});
	file.status = status;
	return file;
}
