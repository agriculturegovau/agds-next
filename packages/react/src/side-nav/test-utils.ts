import { SideNavProps } from './SideNav';

export const defaultTestingProps: SideNavProps = {
	title: 'Lodging your tax return',
	titleLink: '/',
	collapseTitle: 'In this section',
	items: [
		{
			href: '/welcome',
			label: 'Welcome',
		},
		{
			href: '/do-you-need-to-lodge-a-tax-return',
			label: 'Do you need to lodge a tax return?',
		},
		{
			href: '/lodge-online',
			label: 'Lodge online',
			items: [
				{
					href: '/lodge-online/pre-filling',
					label: 'Pre-filling your online tax return',
				},
			],
		},
		{
			href: '/whats-new-for-individuals',
			label: "What's new for individuals",
		},
		{
			href: '/why-you-may-receive-a-tax-bill',
			label: 'Why you may receive a tax bill',
		},
		{
			href: '/in-detail',
			label: 'In detail',
			items: [
				{
					href: '/in-detail/record-keeping',
					label: 'Record keeping',
					items: [
						{
							href: '/in-detail/record-keeping/tax',
							label: 'Keeping your tax records',
						},
						{
							href: '/in-detail/record-keeping/incorrect-amounts',
							label: 'Incorrect amounts',
						},
					],
				},
				{
					href: '/tax-receipt',
					label: 'Tax receipt',
				},
				{
					href: '/pre-fill availability',
					label: 'Pre-fill availability',
				},
			],
		},
	],
	activePath: '/in-detail',
};
