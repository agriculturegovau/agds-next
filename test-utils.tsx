import { PropsWithChildren, ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { Core } from '@ag.ds-next/react/core';
import { theme } from '@ag.ds-next/react/ag-branding';

function AllTheProviders({ children }: PropsWithChildren) {
	return <Core theme={theme}>{children}</Core>;
}

const customRender = (
	ui: ReactElement,
	options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';
export { customRender as render };
