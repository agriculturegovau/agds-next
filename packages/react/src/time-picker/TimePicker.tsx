import { useEffect, useState } from 'react';
import { useCombobox } from 'downshift';
import { type ComboboxProps } from '../combobox/Combobox';
import { ComboboxBase } from '../combobox/ComboboxBase/ComboboxBase';
import {
	type DefaultComboboxOption,
	useComboboxInputId,
} from '../combobox/utils';
import { type TimeFormat } from '../time-input/utils';
import { filterOptions, generateOptions } from './utils';

export function TimePicker({
	id,
	inputRef: inputRefProp,
	interval = 15,
	max = '24:00',
	maxWidth = 'md',
	min = '00:00',
	onChange,
	timeFormat = 'h:mm aaa',
	value,
	...props
}: TimePickerProps) {
	const {
		// We remove this prop so that it can't be passed to ComboboxBase
		// eslint-disable-next-line @typescript-eslint/ban-ts-comment
		// @ts-ignore
		renderItem,
		...consumerProps
	} = props;
	const inputId = useComboboxInputId(id);
	const [options, setOptions] = useState<DefaultComboboxOption[]>(
		generateOptions({ interval, max, min, timeFormat })
	);
	const [inputItems, setInputItems] =
		useState<DefaultComboboxOption[]>(options);

	useEffect(() => {
		setOptions(generateOptions({ interval, max, min, timeFormat }));
	}, [interval, max, min, timeFormat]);

	const combobox = useCombobox<DefaultComboboxOption>({
		items: inputItems,
		selectedItem: value,
		inputId,
		itemToString: (item) => item?.label ?? '',
		onSelectedItemChange: ({ selectedItem = null }) => {
			onChange?.(selectedItem);
		},
		onInputValueChange: ({ inputValue, isOpen }) => {
			inputValue = inputValue?.toLowerCase() ?? '';
			if (isOpen) {
				setInputItems(filterOptions(options, inputValue));
			} else {
				// When the menu is closed by the user, reset the entire options
				// This is common in other Combobox implementations (react-aria, react-select, etc)
				setInputItems(options);
			}
		},
		stateReducer: (state, actionAndChanges) => {
			const { type: actionAndChangesType, changes } = actionAndChanges;
			switch (actionAndChangesType) {
				// Reset the input value when the menu is closed
				case useCombobox.stateChangeTypes.InputBlur:
					return {
						...changes,
						inputValue: state.selectedItem?.label ?? '',
					};
				default:
					return changes;
			}
		},
	});

	return (
		<ComboboxBase
			combobox={combobox}
			inputItems={inputItems}
			inputId={inputId}
			inputRef={inputRefProp}
			isAutocomplete={false}
			maxWidth={maxWidth}
			{...consumerProps}
		/>
	);
}

export type TimePickerProps = Omit<ComboboxProps, 'options' | 'renderItem'> & {
	/* A number of minutes between 1 & 60 to create options at appropriate intervals. */
	interval?: number;
	/* A boolean indicating whether to render the options' loading state. */
	loading?: boolean;
	/* The maximum time for options to end in HH:mm. */
	max?: string;
	/* The minimum time for options to start in HH:mm. */
	min?: string;
	/* The format to render the options. */
	timeFormat?: TimeFormat;
};
