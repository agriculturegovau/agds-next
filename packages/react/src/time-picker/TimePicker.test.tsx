import '@testing-library/jest-dom';
import 'html-validate/jest';
import { axe, toHaveNoViolations } from 'jest-axe';
import userEvent from '@testing-library/user-event';
import { render, cleanup, act, waitFor } from '../../../../test-utils';
import { TimePicker, type TimePickerProps } from './TimePicker';

expect.extend(toHaveNoViolations);

afterEach(cleanup);

function renderTimePicker(props?: Pick<TimePickerProps, 'loading'>) {
	return render(
		<TimePicker
			label="Select a time"
			hint="Start typing to see results"
			{...props}
		/>
	);
}

describe('TimePicker', () => {
	it('renders correctly', () => {
		const { container } = renderTimePicker();
		expect(container).toMatchSnapshot();
	});

	it('renders correctly when loading', async () => {
		const { container } = renderTimePicker({ loading: true });
		const input = container.querySelector('input');
		expect(input).toBeInTheDocument();
		if (!input) return;

		await act(async () => input.click());
		expect(container).toMatchSnapshot();
	});

	it('renders valid HTML with no a11y violations', async () => {
		const { container } = renderTimePicker();
		expect(container).toHTMLValidate({
			extends: ['html-validate:recommended'],
			rules: {
				'no-inline-style': 'off',
				// Our HTML is based off downshift
				'prefer-native-element': 'off',
				// react 18s `useId` break this rule
				'valid-id': 'off',
			},
		});
		await act(async () => {
			expect(await axe(container)).toHaveNoViolations();
		});
	});

	it('filters options', async () => {
		const { container } = renderTimePicker();
		const input = container.querySelector('input');
		expect(input).toBeInTheDocument();
		if (!input) return;

		await act(async () => input.click());
		await waitFor(() => expect(input).toHaveFocus());
		expect(input).toHaveAttribute('aria-expanded', 'true');

		await userEvent.type(input, '1245');
		expect(input.value).toBe('1245');

		const options = container.querySelectorAll('li');
		expect(options.length).toBe(1);
		expect(options[0].textContent).toBe('12:45 pm');
		await userEvent.click(options[0]);
		expect(input.value).toBe('12:45 pm');
	});
});
