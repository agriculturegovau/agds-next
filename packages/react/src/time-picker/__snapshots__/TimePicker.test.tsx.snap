// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimePicker renders correctly 1`] = `
<div>
  <div
    class="css-2ohkpz-boxStyles-FieldContainer"
  >
    <label
      class="css-12ek51e-boxStyles"
      for="combobox-input-:r0:"
      id="downshift-:r1:-label"
    >
      <span
        class="css-1sf8c0k-boxStyles"
      >
        Select a time
      </span>
      <span
        class="css-1ru6lht-boxStyles"
      >
         
        (optional)
      </span>
    </label>
    <span
      class="css-16uykm5-boxStyles"
      id="field-combobox-input-:r0:-hint"
    >
      Start typing to see results
    </span>
    <div
      class="css-8tabia-ComboboxBase"
    >
      <input
        aria-activedescendant=""
        aria-autocomplete="list"
        aria-controls="downshift-:r1:-menu"
        aria-describedby="field-combobox-input-:r0:-hint"
        aria-expanded="false"
        aria-invalid="false"
        aria-labelledby="downshift-:r1:-label"
        aria-required="false"
        autocomplete="off"
        class="css-g1val9-ComboboxBase"
        id="combobox-input-:r0:"
        role="combobox"
        type="text"
        value=""
      />
      <div
        class="css-1xsup3m-boxStyles-ComboboxButtonContainer"
      >
        <button
          aria-controls="downshift-:r1:-menu"
          aria-expanded="false"
          aria-label="Toggle menu"
          class="css-nyx9qe-BaseButton-ComboboxIconButton"
          id="downshift-:r1:-toggle-button"
          tabindex="-1"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="css-1z0c8dw"
            clip-rule="evenodd"
            fill-rule="evenodd"
            focusable="false"
            height="24"
            role="img"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <polyline
              points="6 9 12 15 18 9"
            />
          </svg>
        </button>
      </div>
      <ul
        aria-labelledby="downshift-:r1:-label"
        class="css-udl9lu-boxStyles-Popover"
        id="downshift-:r1:-menu"
        role="listbox"
        style="position: absolute; left: 0px; top: 0px; transform: translate(0px, 0px);"
      />
    </div>
    <div
      aria-hidden="true"
      class="css-1r6wwux-Field"
    />
  </div>
</div>
`;

exports[`TimePicker renders correctly when loading 1`] = `
<div>
  <div
    class="css-2ohkpz-boxStyles-FieldContainer"
  >
    <label
      class="css-12ek51e-boxStyles"
      for="combobox-input-:r3:"
      id="downshift-:r4:-label"
    >
      <span
        class="css-1sf8c0k-boxStyles"
      >
        Select a time
      </span>
      <span
        class="css-1ru6lht-boxStyles"
      >
         
        (optional)
      </span>
    </label>
    <span
      class="css-16uykm5-boxStyles"
      id="field-combobox-input-:r3:-hint"
    >
      Start typing to see results
    </span>
    <div
      class="css-8tabia-ComboboxBase"
    >
      <input
        aria-activedescendant=""
        aria-autocomplete="list"
        aria-controls="downshift-:r4:-menu"
        aria-describedby="field-combobox-input-:r3:-hint"
        aria-expanded="true"
        aria-invalid="false"
        aria-labelledby="downshift-:r4:-label"
        aria-required="false"
        autocomplete="off"
        class="css-g1val9-ComboboxBase"
        id="combobox-input-:r3:"
        role="combobox"
        type="text"
        value=""
      />
      <div
        class="css-1xsup3m-boxStyles-ComboboxButtonContainer"
      >
        <button
          aria-controls="downshift-:r4:-menu"
          aria-expanded="true"
          aria-label="Toggle menu"
          class="css-nyx9qe-BaseButton-ComboboxIconButton"
          id="downshift-:r4:-toggle-button"
          tabindex="-1"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="css-1z0c8dw"
            clip-rule="evenodd"
            fill-rule="evenodd"
            focusable="false"
            height="24"
            role="img"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <polyline
              points="18 15 12 9 6 15"
            />
          </svg>
        </button>
      </div>
      <ul
        aria-labelledby="downshift-:r4:-label"
        class="css-1569u13-boxStyles-Popover"
        id="downshift-:r4:-menu"
        role="listbox"
        style="position: absolute; left: 0px; top: 0px; transform: translate(1px, 8px); max-height: 195px; width: 0px;"
      >
        <li
          class="css-g0k1ob-ComboboxListItem"
          tabindex="-1"
        >
          <div
            class="css-3ifbi3-boxStyles"
          >
            <span
              aria-atomic="false"
              aria-label="Loading options"
              class="css-uwg2j1-boxStyles"
              role="alert"
            >
              <span
                aria-hidden="true"
                class="css-1pzu52k-boxStyles-LoadingDots"
              />
              <span
                aria-hidden="true"
                class="css-1v3ghdi-boxStyles-LoadingDots"
              />
              <span
                aria-hidden="true"
                class="css-7kx10c-boxStyles-LoadingDots"
              />
            </span>
            <span
              class="css-1e881ou-boxStyles"
            >
              Loading
            </span>
          </div>
        </li>
      </ul>
    </div>
    <div
      aria-hidden="true"
      class="css-1r6wwux-Field"
    />
  </div>
</div>
`;
